ngraph/core/src/op/gather_elements.cpp

```diff
@@ -0,0 +1,128 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/op/gather_elements.hpp"
+#include "ngraph/shape.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+// ------------------------------ V6 ------------------------------
+
+NGRAPH_RTTI_DEFINITION(op::v6::GatherElements, "GatherElements", 6);
+
+op::v6::GatherElements::GatherElements(const Output<Node>& data,
+                                       const Output<Node>& indices,
+                                       const int64_t axis)
+    : Op({data, indices})
+    , m_axis(axis)
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::v6::GatherElements::validate_and_infer_types()
+{
+    const auto& data_type = get_input_element_type(0);
+    const auto& indices_type = get_input_element_type(1);
+
+    NODE_VALIDATION_CHECK(this,
+                          indices_type == element::Type_t::i32 ||
+                              indices_type == element::Type_t::i64,
+                          "indices must be of int32 or int64 type. But instead got: ",
+                          indices_type);
+
+    const auto& data_pshape = get_input_partial_shape(0);
+    const auto& indices_pshape = get_input_partial_shape(1);
+    auto data_rank = data_pshape.rank();
+    auto indices_rank = indices_pshape.rank();
+
```

Comment:
Isn't it what documentation says?
```suggestion
    set_output_type(0, indices_pshape);
```

Reply:
Agree. I added set_out_type here close to the beginning.